common/persistence/serialization/interfaces.go

```diff
@@ -35,267 +35,267 @@ import (
-		CreateRequestID        *string
-		DomainName             *string
-		WorkflowTypeName       *string
-		ParentClosePolicy      *int32
+		StartedEventEncoding   string
+		CreateRequestID        string
+		DomainName             string
+		WorkflowTypeName       string
+		ParentClosePolicy      int32
 	}
 
 	// SignalInfo blob in a serialization agnostic format
 	SignalInfo struct {
-		Version               *int64
-		InitiatedEventBatchID *int64
-		RequestID             *string
-		Name                  *string
+		Version               int64
+		InitiatedEventBatchID int64
+		RequestID             string
+		Name                  string
 		Input                 []byte
 		Control               []byte
 	}
 
 	// RequestCancelInfo blob in a serialization agnostic format
 	RequestCancelInfo struct {
-		Version               *int64
-		InitiatedEventBatchID *int64
-		CancelRequestID       *string
+		Version               int64
+		InitiatedEventBatchID int64
+		CancelRequestID       string
 	}
 
 	// TimerInfo blob in a serialization agnostic format
 	TimerInfo struct {
-		Version         *int64
-		StartedID       *int64
-		ExpiryTimestamp *time.Time
-		TaskID          *int64
+		Version         int64
+		StartedID       int64
+		ExpiryTimestamp time.Time
+		TaskID          int64
 	}
 
 	// TaskInfo blob in a serialization agnostic format
 	TaskInfo struct {
-		WorkflowID       *string
+		WorkflowID       string
 		RunID            UUID
-		ScheduleID       *int64
-		ExpiryTimestamp  *time.Time
-		CreatedTimestamp *time.Time
+		ScheduleID       int64
+		ExpiryTimestamp  time.Time
+		CreatedTimestamp time.Time
 	}
 
 	// TaskListInfo blob in a serialization agnostic format
 	TaskListInfo struct {
-		Kind            *int16
-		AckLevel        *int64
-		ExpiryTimestamp *time.Time
-		LastUpdated     *time.Time
+		Kind            int16
+		AckLevel        int64
+		ExpiryTimestamp time.Time
+		LastUpdated     time.Time
 	}
 
 	// TransferTaskInfo blob in a serialization agnostic format
 	TransferTaskInfo struct {
 		DomainID                UUID
-		WorkflowID              *string
+		WorkflowID              string
 		RunID                   UUID
-		TaskType                *int16
+		TaskType                int16
 		TargetDomainID          UUID
-		TargetWorkflowID        *string
+		TargetWorkflowID        string
 		TargetRunID             UUID
-		TaskList                *string
-		TargetChildWorkflowOnly *bool
-		ScheduleID              *int64
-		Version                 *int64
-		VisibilityTimestamp     *time.Time
+		TaskList                string
+		TargetChildWorkflowOnly bool
+		ScheduleID              int64
+		Version                 int64
+		VisibilityTimestamp     time.Time
 	}
 
 	// TimerTaskInfo blob in a serialization agnostic format
 	TimerTaskInfo struct {
 		DomainID        UUID
-		WorkflowID      *string
+		WorkflowID      string
 		RunID           UUID
-		TaskType        *int16
+		TaskType        int16
 		TimeoutType     *int16
```

Comment:
The default value for `TimeoutType` doesn't make sense. For example, user timers previously have this field set to `nil`, but if we change it back to int16, the default value doesn't make sense to user timers. 
Ideally, for all enum types, the default values should be an invalid value.